pkg/cmd/dictionary/entries/clear/clear.go

```diff
@@ -0,0 +1,143 @@
+
+// NewClearCmd creates and returns a clear command for dictionnaries' entries.
+func NewClearCmd(f *cmdutil.Factory, runF func(*ClearOptions) error) *cobra.Command {
+	var confirm bool
+	cs := f.IOStreams.ColorScheme()
+
+	opts := &ClearOptions{
+		IO:           f.IOStreams,
+		Config:       f.Config,
+		SearchClient: f.SearchClient,
+	}
+	cmd := &cobra.Command{
+		Use:       "clear {<dictionnary>... | --all} [--confirm]",
+		Args:      cobra.OnlyValidArgs,
+		ValidArgs: DictionaryNames(),
+		ValidArgsFunction: func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+			return DictionaryNames(), cobra.ShellCompDirectiveNoFileComp
+		},
+		Short: "Clear dictionary entries",
+		Long: heredoc.Docf(`
+			This command deletes all entries from the specified %s dictionnaries.
+		`, cs.Bold("custom")),
+		Example: heredoc.Doc(`
+			# Delete all entries from the "stopword" dictionnary
+			$ algolia dictionary entries clear stopword
+
+			# Delete all entries from the "stopword" and "plural" dictionnaries
+			$ algolia dictionary entries clear stopword plural
+
+			# Delete all entries from all dictionnaries
+			$ algolia dictionary entries clear --all
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if opts.All && len(args) > 0 || !opts.All && len(args) == 0 {
+				return cmdutil.FlagErrorf("Either specify dictionnaries' names or use --all to clear all dictionnaries")
+			}
+
+			if opts.All {
+				opts.Dictionnaries = []search.DictionaryName{search.Stopwords, search.Plurals, search.Compounds}
+			} else {
+				opts.Dictionnaries = make([]search.DictionaryName, len(args))
+				for i, dictionnary := range args {
+					opts.Dictionnaries[i] = search.DictionaryName(dictionnary)
+				}
+			}
+
+			if !confirm {
+				if !opts.IO.CanPrompt() {
+					return cmdutil.FlagErrorf("--confirm required when non-interactive shell is detected")
+				}
+				opts.DoConfirm = true
+			}
+
+			if runF != nil {
+				return runF(opts)
+			}
+
+			return runClearCmd(opts)
+		},
+	}
+
+	cmd.Flags().BoolVarP(&confirm, "confirm", "y", false, "skip confirmation prompt")
+	cmd.Flags().BoolVarP(&opts.All, "all", "a", false, "clear all dictionnaries")
+
+	return cmd
+}
+
+// runClearCmd executes the clear command
+func runClearCmd(opts *ClearOptions) error {
+	client, err := opts.SearchClient()
+	if err != nil {
+		return err
+	}
+
+	dictionnaries := opts.Dictionnaries
+	dictionnariesNames := make([]string, len(dictionnaries))
+	for i, dictionnary := range dictionnaries {
+		dictionnariesNames[i] = string(dictionnary)
+	}
+
+	if opts.DoConfirm {
+		var confirmed bool
+		err = prompt.Confirm(fmt.Sprintf("Clear all entries from %s dictionary?", utils.SliceToReadableString(dictionnariesNames)), &confirmed)
+		if err != nil {
+			return fmt.Errorf("failed to prompt: %w", err)
+		}
+		if !confirmed {
+			return nil
+		}
+	}
+
+	for _, dictionnary := range dictionnaries {
+		_, err = client.ClearDictionaryEntries(dictionnary)
```

Comment:
Wondering if we should take care of reverting operations if there's an error? It's more a thought for the whole CLI, maybe we should define this together

Reply:
I don't think we want. This is out of scope of the CLI IMHO. Also, in order to revert an operation, you will need to store the objects somewhere (on file? In another app/indice?). And how to make 100% sure those objects can be later retrieved? I honestly don't think we want to go on this path 🤔

Reply:
Got it, we had the same discussion when we wanted to do `algolia app migrate` 👍